src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioAnalyzer.cs

```diff
@@ -55,7 +55,9 @@ public AnalyzerReference GetReference()
             {
                 if (File.Exists(_fullPath))
                 {
-                    _analyzerReference = new AnalyzerFileReference(_fullPath, _loader);
+                    // Pass down a custom loader that will ensure we are watching for file changes once we actually load the assembly.
+                    var assemblyLoaderForFileTracker = new AnalyzerAssemblyLoaderThatEnsuresFileBeingWatched(this);
+                    _analyzerReference = new AnalyzerFileReference(_fullPath, assemblyLoaderForFileTracker);
```

Comment:
This means that the watcher's subscription won't be created if `GetReference` is called when the file doesn't exist.